### PR TITLE
[SD-1319] : Fix: Status switch is saving inverted state, when status is ...

### PR DESCRIPTION
**Pull request by sjunaid**
...not changed and Save.

### CODE REVIEW

**Comment by nidzix** [comment]
I'm not sure that we need new `directive` for this. If we do, than code is :ok:. 
:+1:

**Comment by sjunaid** [comment]
@nidzix , If we change the existing switch for inverted functionality, then we need to test that switch for both type of functionality all where its using. that's why i preferred to create a separate switch for inverted functionality.

File: app/scripts/superdesk/directives/sdCheck.js [hunk 1]
```diff
@@ -87,6 +87,46 @@ define(['angular'], function(angular) {
...
+                    '<span class="sd-toggle">',
+                    '<span class="inner"></span>',
+                    '</span>'
+                ].join(''),
+                link: function($scope, element, attrs, ngModel) {
+                       ngModel.$render = function() {
+                        render(element, ngModel.$viewValue);
+                    };
+
+                    $scope.$watch(attrs.ngModel, function() {
```
Comment (petrjasek) [inline app/scripts/superdesk/directives/sdCheck.js]: this will never change so is there a need for `$watch` ?

**Comment by sjunaid** [comment]
Yes, this $watch renders switch ON/OFF depending upon when user clicks on switch,in this case it renders inverted behavior of switch.

**Comment by petrjasek** [comment]
my point is that `attrs.ngModel` is a value or reference but it will be evaluated only once and never change. if it is changing it should be like:
```
$scope.$watch(function() { return attrs.ngModel; }
```

**Pull request merged**